Framework/Kernel/src/ErrorReporter.cpp

```diff
@@ -99,10 +99,8 @@ std::string ErrorReporter::generateErrorMessage() {
 
   message["exitCode"] = m_exitCode;
 
-  if (m_share) {
-    message["email"] = m_email;
-    message["name"] = m_name;
-  }
+  message["email"] = m_email;
```

Comment:
Isn't this now sharing the name/email when it has not been asked too?

Reply:
I just noticed that clang says `m_share` is not used any more. Does this mean the share/no share stuff is handled differently now?

Reply:
@martyngigg 
This is due to the change made where the share only non identifying information is grayed out if anything is entered in the name or email boxes. This means that for this case the name and email we are sharing will always be empty strings. This does somewhat remove the need for m_share at all. The change was originally made becuase the web service was returning an error code if the post request it received didn't have a name and email field. 

I think there are two solutions which would be better than this time allowing
* Keep the if statement and explicitly set name and email to empty strings to reduce chance of future changes breaking confidentiality. 
* Revert this change entirely and update the web service to be able to deal with post requests which do not contain name and email.

Reply:
At this point I think I would opt for option 1. As you say, it's not guaranteed that the name/email is blank so I think forcing those fields blank when they are not requested indicates to future developers that there is a different situation to deal with here.